PerformanceTests/JetStream2/JetStreamDriver.js

```diff
@@ -640,75 +660,158 @@ class Benchmark {
+                }));
+            }
+        }
 
-                ++counter.loadedFiles;
-                var statusElement = document.getElementById("status");
-                statusElement.innerHTML = `Loading ${counter.loadedFiles} of ${counter.totalFiles} ...`;
-            });
-        }));
+        JetStream.counter.totalResources += promises.length;
+        return Promise.all(promises);
     }
 
-    fetchResources() {
-        if (this._resourcesPromise)
-            return this._resourcesPromise;
+    async retryPrefetchResource(type, prop, file) {
+        let counter = JetStream.counter;
+        var blobData = JetStream.blobDataCache[file];
+        if (blobData.blob) {
+            // The same preload blob may be used by multiple subtests. Though the blob is already loaded,
+            // we still need to check if this subtest failed to load it before. If so, handle accordingly.
+            if (type == "preload") {
+                if (this.failedPreloads && this.failedPreloads[blobData.prop]) {
+                    this.failedPreloads[blobData.prop] = false;
+                    this.preloads.push([ blobData.prop, blobData.blobURL ]);
+                    counter.failedPreloadResources--;
+                }
+            }
+            return !counter.failedPreloadResources && counter.loadedResources == counter.totalResources;
+        }
 
-        let filePromises = !isInBrowser ? this.plan.files.map((file) => fileLoader.load(file)) : [];
-        let preloads = [];
-        let preloadVariableNames = [];
+        // Retry fetching the resource.
+        JetStream.loadCache[file] = null;
+        await this.loadBlob(type, prop, file, false).then((blobData) => {
+            if (!window.allIsGood)
+                return;
+            if (blobData.type == "preload")
+                this.preloads.push([ blobData.prop, blobData.blobURL ]);
+            this.updateCounter();
+        });
+
+        if (!blobData.blob) {
+            window.allIsGood = false;
+            throw new Error("Fetch failed"); 
+        }
+
+        return !counter.failedPreloadResources && counter.loadedResources == counter.totalResources;
+    }
+
+    async retryPrefetchResourcesForBrowser() {
+        if (!isInBrowser)
+            return;
 
-        if (isInBrowser && this.plan.preload) {
+        let counter = JetStream.counter;
+        for (let resource of this.plan.files) {
+            let allDone = await this.retryPrefetchResource("file", null, resource);
+            if (allDone)
+                return true; // All resources loaded, nothing more to do.
+        }
+
+        if (this.plan.preload) {
             for (let prop of Object.getOwnPropertyNames(this.plan.preload)) {
-                preloadVariableNames.push(prop);
-                preloads.push(this.plan.preload[prop]);
+                let resource = this.plan.preload[prop];
+                let allDone = await this.retryPrefetchResource("preload", prop, resource);
```

Comment:
Are we sure that we don't want to `Promise.all(this.plan.files.map())` here?

Reply:
Ditto.